Feat: Quiet is trying to connect... and Sending... on messages when not connected to other peers #1706

islathehut opened the pull request:
* Show a spinner with the message `Quiet is trying to connect...` in message pane when no other peers are connected and community has peers
* Show `Sending...` with animated ellipsis on messages when no other peers are connected and community has peers
* Update opacity of messages (text, math, file and image), usernames, and avatars in message pane when no other peers are connected and community has peers
* Adds selectors/reducers for getting peer data, final peer disconnect time and last time the user connected to the community
* Adds `AnimatedEllipsis` class for showing text followed by an animated ellipsis
* Adds tests
* Adds reliability features to E2E tests
* Allows logs from desktop renderer and state-manager to appear in console with other logs (e.g. backend)

https://www.figma.com/file/frxdNd42wV8kgXoiBExoqR/User-and-network-status?type=design&node-id=716-24467&mode=design&t=HRARHrzbOeNjpl8o-0

### Pull Request Checklist

- [x] I have linked this PR to a related GitHub issue.
- [x] I have added a description of the change (and Github issue number, if any) to the root [CHANGELOG.md](https://github.com/TryQuiet/quiet/blob/develop/CHANGELOG.md).

### (Optional) Mobile checklist

Please ensure you completed the following checks if you did any changes to the mobile package:

- [ ] I have run e2e tests for mobile
- [ ] I have updated base screenshots for visual regression tests

islathehut committed:
Add "Quiet is trying to connect..." panel

islathehut committed:
Most of the way to "Sending..." and graying out logic

islathehut committed:
Sending... and graying out text, files, and images when unsent as well as fixing grouping of messages with unsent behavior

islathehut committed:
Don't use this logic on fresh communities

islathehut commented:
Implements #1706

islathehut committed:
Remove commented out code

islathehut committed:
Don't change display on info message

islathehut committed:
Formatting

islathehut committed:
Better peer disconnect/connect handling

islathehut committed:
Update desktop tests

islathehut committed:
Merge branch 'develop' into feat/1706-connection-status-desktop

islathehut committed:
Linting

islathehut committed:
Remove unused luxon import

islathehut committed:
Downgrade block scoping babel plugin to fix regression in their code

islathehut committed:
Merge branch 'develop' into feat/1706-connection-status-desktop

islathehut committed:
test updates

islathehut committed:
Merge branch 'develop' into feat/1706-connection-status-desktop

islathehut committed:
Retries and more stable tests

islathehut committed:
Fix tests around registration messages

islathehut committed:
Fix bad conditions

islathehut committed:
Update snapshots with new data-testids

islathehut committed:
Wrong user

islathehut committed:
More test fixes

islathehut committed:
More test fixes

islathehut committed:
Fix package changes I accidentally pushed

islathehut committed:
Update snapshot

islathehut committed:
Why is this now failing?

islathehut committed:
Fix test

islathehut committed:
Update tests

islathehut committed:
Run all rtl tests again

islathehut committed:
E2E debugging

islathehut committed:
Get logs from state manager and desktop renderer in node console

islathehut committed:
Fix backwards compatibility tests

islathehut committed:
Fix issue with cypress not recognizing console constructor

islathehut committed:
Fix imports

Commit:
Delete test.log that was added by mistake

islathehut committed:
Update changelogs

islathehut committed:
Update CHANGELOG.md

islathehut committed:
Mild logger refactor

islathehut committed:
Pull lfs in checkout

EmiM commented on packages/desktop/src/renderer/components/Channel/ChannelComponent.tsx:
```diff
@@ -241,6 +286,33 @@ export const ChannelComponent: React.FC<ChannelComponentProps & UploadFilesPrevi
             duplicatedUsernameModalHandleOpen={duplicatedUsernameModalHandleOpen}
           />
         </ChannelMessagesWrapperStyled>
+        <Grid
```
I'm wondering if maybe all this new code in ChannelComponent could be a completely new component. I mean it has its own purpose and it can have its own separate place in storybook.
Also (for the sake of clarity) its props could be grouped in Channel.tsx similar to uploadFilesPreviewProps or fileActionsProps.

islathehut replied:
Merged the code

EmiM commented on packages/desktop/src/renderer/components/ChangeUsername/ChangeUsername.component.tsx:
```diff
@@ -16,6 +16,7 @@ import { TextInput } from '../../forms/components/textInput'
 import { userNameField } from '../../forms/fields/createUserFields'
 
 import { parseName } from '@quiet/common'
+import { defaultLogger } from '../../logger'
```
Could we just use 'logger' instead of 'defaultLogger'? I think the fact that the logger is 'default' should be implicit.

islathehut replied:
This is a placeholder.  Eventually I'd like to update this to have named loggers like we have in every other module (e.g. `desktop:renderer:someModule`) but for now this is the default.

EmiM commented on packages/state-manager/src/sagas/publicChannels/publicChannels.selectors.ts:
```diff
@@ -226,17 +243,29 @@ export const currentChannelMessagesMergedBySender = createSelector(
+        const hasConnectedPeers = connectedPeers.length > 0
+        const peersDisconnectedRecently = allPeersDisconnectedAt != null && allPeersDisconnectedAt < message.createdAt
+        const noPeersThisSession = allPeersDisconnectedAt == null && connectedPeers.length > 0
+        const isUnsent =
```
I can see a very similar code in BasicMessage.tsx. One could easily miss one of those places when updating the 'unsent' logic. Maybe we could refactor it to use the same function? It can also be easier testable this way.

islathehut replied:
Done.

EmiM commented on packages/desktop/src/renderer/components/Channel/File/FileComponent/FileComponent.test.tsx:
```diff
@@ -38,13 +38,14 @@ describe('FileComponent', () => {
           downloadState: DownloadState.Ready,
           downloadProgress: undefined,
         }}
+        isUnsent={false}
```
I think we are missing components tests for case where `isUnsent={true}`.

islathehut replied:
Added component tests

islathehut committed:
Share code for determining if message is unsent

islathehut committed:
Updates from PR comments

Commit:
Merge branch 'develop' into feat/1706-connection-status-desktop

islathehut committed:
Add more desktop tests

Commit:
Fix: Not initializing on reconnection when general channel was deleted (#2334) (#2400)

* Fix issue with users joining a community where the general channel was deleted while they were offline

* Update e2e tests to include case for this bug

* Update CHANGELOG.md

islathehut committed:
Merge branch 'develop' into feat/1706-connection-status-desktop

islathehut committed:
Delete files that were deleted on develop

Commit:
Publish

 - @quiet/desktop@2.2.0-alpha.0
 - @quiet/mobile@2.2.0-alpha.0

Commit:
Update packages CHANGELOG.md

islathehut committed:
Merge from 2.2.0

Commit:
Merge branch '2.2.0' into feat/1706-connection-status-desktop

islathehut committed:
Add some extra test cases

islathehut committed:
Remove tor initialized emit I added and add some logging

Commit:
fix: hanging process functionality killing wrong processes (#2419)

When using backend data directories like `Quietdev`, `Quietdev1`, `Quietdev2',
restarting the process using `Quietdev` can result in that process killing the
other backend processes due to the grep command used for finding hanging
processes. Example:

```
user (Quietdev1):
desktop:main:main Forked backend, PID: 214543

owner (Quietdev):
Found 1 hanging backend process(es) with pid(s) 214543. Killing...
```

islathehut committed:
Merge branch '2.2.0' into feat/1706-connection-status-desktop

Commit:
fix: Adds a quick fix for the iOS sync issue after suspend (#2414)

iOS appears to not be syncing correctly after resuming from suspend.
It looks like this is due to the libp2p auto-dialler getting stuck
connecting to the Tor HTTP tunnel port which is replaced after
suspend/resume. This commit enables the WebsocketOverTor transport to
use the `dialTimeout` correctly. But still this means the current
dialer waits for `dialTimeout` to try another peer, and we currently
have `dialTimeout` set to 2 minutes. Ideally, we can find a better
approach for closing any sockets when suspending the app or detecting
when we are trying to connect to a port that nothing is listening on,
if that's possible.

islathehut committed:
Merge branch '2.2.0' into feat/1706-connection-status-desktop

islathehut committed:
stashing

islathehut committed:
stashing

Commit:
fix: Remove unused dmPublicKey to prevent UI delay during joining (#2392)

Commit:
refactor: Remove SET_COMMUNITY_METADATA event (#2408)

islathehut committed:
Merge from 2.2.0

islathehut committed:
Test fix and remove logs

islathehut committed:
Better peer sorting and updated initial diallng

islathehut committed:
Fix tests

islathehut committed:
Fix snapshots

islathehut committed:
Revert "Better peer sorting and updated initial diallng"

This reverts commit 98435ef8a69630f63e080b88fe112d086928711f.

islathehut committed:
Merge from develop

islathehut committed:
revert merge from 2.2.0

islathehut committed:
Revert "Merge branch '2.2.0' into feat/1706-connection-status-desktop"

This reverts commit d3681706d30787bc1b8b7f722991829da6d41efa, reversing
changes made to b45a1ec717ce15ec1c0fac202b36780f138adb14.

islathehut committed:
Revert "Merge branch '2.2.0' into feat/1706-connection-status-desktop"

This reverts commit b45a1ec717ce15ec1c0fac202b36780f138adb14, reversing
changes made to e1fd05917301f45ad87308c53589396904b08b53.

islathehut committed:
Revert "Merge branch '2.2.0' into feat/1706-connection-status-desktop"

This reverts commit 1f82559b22bfbc38e4509709aba609ef09feaad9, reversing
changes made to 6e88e98eb94c3c38cf229e444bc88ba117f4311e.

islathehut committed:
Update Changelog

islathehut committed:
merge from develop
